### DISCUSSION TITLE
How I use Yasolr

### DISCUSSION THREAD

**Discussion by benoit-cty** [mathieucarbou/YaSolR#22] [Announcements]
Hello,
My solar routing is working like this :
- Yasolr in the electric panel with a jsy sensor
- mqtt in Home Assistant
- A custom automation to compute the level for the heater
- A custom esp32 code to drive a robodyn dimmer connected to the heater
- An esphome with a 18b20 to measure the temperature of the water.

What's your plan for Yasolr to support this workflow (maybe simply it)?

My heater is far away from the electric panel.

Thanks again for your projet!

**Comment by mathieucarbou** [comment]
Hello, yes I plan to support remoting once everything will be in place. 
But it will be the way around: 
- yasolr close to the heater (with dimmer, relays, temp sensor, etc)
- the "source" (jsy in your case) will be remote (I plan to use espnow)
Right now, this setup is already possible if you use an MQTT Topic to read the grid power and work some automations to manually control the relays, dimmers, etc.

Note that the project is still in dev, lot of things to do still, and I am going to write a more complete dimmer library because none exist which supports both phase control, burst mode and new types of ZCD circuit (like this one: https://www.pcbway.com/project/shareproject/Zero_Cross_Detector_Pro_78fbfc76.html) which are quite recent, and where the pulse is really precise.

**Reply by benoit-cty** [reply]
Closing the discussion as it works now.
